slsim/Halos/halos_lens.py

```diff
@@ -0,0 +1,1896 @@
+from lenstronomy.Util.util import make_grid
+import numpy as np
+import astropy.units as u
+from lenstronomy.LensModel.lens_model import LensModel
+from lenstronomy.Cosmo.lens_cosmo import LensCosmo
+import warnings
+import math
+import time
+import multiprocessing
+from collections.abc import Iterable
+
+
+def deg2_to_cone_angle(solid_angle_deg2):
+    """Convert solid angle from square degrees to half cone angle in radians.
+
+    This function translates a solid angle, specified in square degrees, into the
+    corresponding half cone angle expressed in radians. This conversion is essential for
+    applications involving angular measurements in astronomy, particularly in lensing
+    calculations where the geometry of observations is defined in terms of cone angles.
+
+    :param solid_angle_deg2: Solid angle in square degrees to be converted.
+    :type solid_angle_deg2: float
+    :return: The half cone angle in radians equivalent to the input solid angle.
+    :rtype: float :note: The conversion utilizes the relationship between solid angles
+        in steradians and the apex angle of a cone, facilitating a direct transition
+        from square degrees to radians.
+    """
+
+    solid_angle_sr = solid_angle_deg2 * (np.pi / 180) ** 2
+    theta = np.arccos(1 - solid_angle_sr / (2 * np.pi))  # rad
+    return theta
+
+
+def cone_radius_angle_to_physical_area(radius_rad, z, cosmo):
```

Comment:
this could be moved to an utility file as it might be useful for other routines as well